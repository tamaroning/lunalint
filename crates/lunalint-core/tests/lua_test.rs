mod helper;

use std::path::PathBuf;
use strip_ansi_escapes::strip_str;

use insta::Settings;

macro_rules! ident_to_str {
    ($ident:ident) => {
        stringify!($ident).replace("_", "-")
    };
}

macro_rules! lua_test {
    ($name:ident) => {
        #[test]
        fn $name() {
            let name = ident_to_str!($name);
            let path = PathBuf::from(format!("tests/lua/{}.lua", name));
            let out = strip_str(helper::run_linter(&path));
            let mut settings = Settings::new();
            settings.set_prepend_module_to_snapshot(false);
            settings.set_omit_expression(true);
            settings.bind(|| insta::assert_snapshot!(out));
        }
    };
}

lua_test!(undefined_global);
lua_test!(parse_error);
