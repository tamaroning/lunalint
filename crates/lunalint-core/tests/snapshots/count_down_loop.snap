---
source: crates/lunalint-core/tests/lua_test.rs
---
Error: Count down loop which never reaches end (count-down-loop)
    ╭─[tests/lua/count-down-loop.lua:35:9]
    │
 35 │ for i = 10, 1 do
    │         ──┬──  
    │           ╰──── Did you mean `10, 1 , -1`?
    │ 
    │ Help: for further information visit https://luals.github.io/wiki/diagnostics/#count-down-loop
────╯
Error: Count down loop which never reaches end (count-down-loop)
    ╭─[tests/lua/count-down-loop.lua:39:9]
    │
 39 │ for i = 10, -0x99999999999999999999999 do
    │         ───────────────┬──────────────  
    │                        ╰──────────────── Did you mean `10, -0x99999999999999999999999 , -1`?
    │ 
    │ Help: for further information visit https://luals.github.io/wiki/diagnostics/#count-down-loop
────╯
Error: Count down loop which never reaches end (count-down-loop)
    ╭─[tests/lua/count-down-loop.lua:44:9]
    │
 44 │ for i = 0x99999999999999999999999, 0 do
    │         ──────────────┬─────────────  
    │                       ╰─────────────── Did you mean `0x99999999999999999999999, 0 , -1`?
    │ 
    │ Help: for further information visit https://luals.github.io/wiki/diagnostics/#count-down-loop
────╯
